### PR TITLE
fix(bug): over eager fast-levenshtein comparison with URLs containing open

### PR DESCRIPTION
**Pull request by sambacha**
Signed-off-by: sam bacha <sam@manifoldfinance.com>

### CODE REVIEW

**Comment by sambacha** [comment]
Updated `DEFAULT_TOLERANCE` to `4.5` , would like to know how best to compare before and after

**Comment by sambacha** [comment]
Also version bump, minor or patch?

**Comment by sambacha** [comment]
this closes:

https://github.com/MetaMask/eth-phishing-detect/issues/5495
https://github.com/MetaMask/eth-phishing-detect/issues/5500
https://github.com/MetaMask/eth-phishing-detect/issues/5502
https://github.com/MetaMask/eth-phishing-detect/issues/5492
https://github.com/MetaMask/eth-phishing-detect/issues/5494
https://github.com/MetaMask/eth-phishing-detect/issues/5479
https://github.com/MetaMask/eth-phishing-detect/issues/5496
https://github.com/MetaMask/eth-phishing-detect/issues/5496
https://github.com/MetaMask/eth-phishing-detect/issues/5495

**Comment by sambacha** [comment]
@davidwgrossman / @409H , any feedback is welcomed, can close this PR if y'all prefer the aggressive tolerance setting and have addresses added after the fact.

File: src/config.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "version": 2,
   "tolerance": 2,
   "fuzzylist": [
+    "openmev.org",
```
Comment (Gudahtt) [inline src/config.json]: Could you move this to a separate PR? So that we can consider the functional changes separately from the config changes.

File: package.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
     "tap-summary/tap-out/trim": "^0.0.3"
   },
   "dependencies": {
-    "fast-levenshtein": "^2.0.6"
+    "fast-levenshtein": "^3.0.0"
```
Comment (Gudahtt) [inline package.json]: What motivated this update?

File: src/config.json (hunk 1 above)
Comment (sambacha) [reply src/config.json]: sure, in fact I opened a PR specifically for this when I saw PR's being merged that only added to the list.

**Comment by davidwgrossman** [comment]
> @davidwgrossman / @409H , any feedback is welcomed, can close this PR if y'all prefer the aggressive tolerance setting and have addresses added after the fact.

My concern with bumping up the tolerance is that false positives will likely increase (which is an ongoing issue for maintenance of this repository).

File: src/config.json (hunk 1 above)
Comment (Gudahtt) [reply src/config.json]: It looks like there's still a config update in this PR. Also you've updated the package version and license. Please remove those changes as well.

**Comment by sambacha** [comment]
> > @davidwgrossman / @409H , any feedback is welcomed, can close this PR if y'all prefer the aggressive tolerance setting and have addresses added after the fact.
> 
> My concern with bumping up the tolerance is that false positives will likely increase (which is an ongoing issue for maintenance of this repository). If we're looking to decrease false positives, then maybe we should decrease the threshold?

Building off this, how well do you think a heuristic of taking
```
( [DNS Registration Date of Whitelisted FQDN <Modified Julian Date>] + [# Days] ) = DNS Registration Heuristic
```

Compare the DNS Registration Information for the URL (in Mod. Julian Date), if it's greater than the value of the [DNS Reg. Heuristic] by some constant (e.g. 21 days), flag it for further analysis.

Modified Julian Date because we want to compare `number`/`int`.

Example:
```
Sept, 29th, 2021 in Modified Julian Date is = 59486
21 days + 59486 = 59507 = 10-20-2021
```

**Domains registered after 10-20-2021**, would be flagged for fuzz analysis.
**Domains registered before 09-08-2021** would be considered exempt, or could be analyzed with a looser tolerance/sensitivity setting

What are the chances of a phishing URL targeting a domain name being pre-registered before the canonical URL in question? I would wager few and far between.


Back to your point @davidwgrossman , I 100% see your point, what in your estimation would be a saner default for the threshold setting? Any sort of testing you can recommend so I can compare results at different levels? Thank you.

**Comment by 409H** [comment]
The heuristic of getting the DNS registration date is nice - would it be feasible for this PR or are you discussing potentials for a PR in the future. Wondering to see if the commits for this PR is frozen so we can properly give a final review

**Comment by davidwgrossman** [comment]
I'm going to close this PR for now as it is a bit stale. If someone would like to start the conversation again, we can talk more about it in `/issues`
